src/stpipe/step.py

```diff
@@ -431,6 +463,42 @@ def run(self, *args):
             step_result = None
 
             self.log.info("Step %s running with args %s.", self.name, args)
+
+            # Check for explicit disable for CRDS parameters
+            disable_crds_steppars = kwargs.pop("disable_crds_steppars", None)
+
+            # Get parameters from user
+            parameters = None
+            if kwargs:
+                # Unset initialization for any provided override keywords
+                for key in kwargs:
+                    if key in self._initialized:
+                        self._initialized[key] = False
+                parameters = kwargs
+
+            # Get parameters from CRDS
+            if self._validate_kwds:
+                # Get the first filename, if available
+                filename = None
+                if len(args) > 0:
+                    filename = args[0]
+
+                # Build config from CRDS + user keywords
+                try:
+                    parameters, _ = self.build_config(
```

Comment:
Can this be skipped entirely if `disable_crds_steppars==True`? It appears to be opening the input file and doing nothing with it in that case.

Reply:
Good point. I'll fix that.